core/eval.go

```diff
@@ -261,6 +261,11 @@ func evalGET(args []string, store *Store) []byte {
 		return RespNIL
 	}
 
+	// If the object exists, check if it is a set object.
+	if err := assertType(obj.TypeEncoding, ObjTypeSet); err == nil {
```

Comment:
Shall L270 i.e `switch _, oEnc := ExtractTypeEncoding(obj); oEnc` shouldn't handle this case?
Any specific reasons to introduce this check here?

Reply:
The existing switch statement does not return the message similar to the Redis implementation.

Reply:
In `default` it's having return error as `diceerrors.WrongTypeErr`.
I'd suggest having consistent way of having determining object encoding instead of multiple assert blocks.
@soumya-codes thoughts?

Reply:
I think it checks the encoding and not the type of the value stored. GET currently expects only int and string as the value of the object (which is inline with redis implementation). In redis, the get functions only return strings. For now, the encoding of the SET values are similar to the ObjEncodingRaw's value, which is why it matches the second case. 

I am thinking following:
1. Create two new encodings for sets:

Current
```
var ObjEncodingHT uint8 = 0
```

New
```
var ObjEncodingSetInt uint8 = 11
var ObjEncodingSetStr uint8 = 12
```
2. Update the asserts for this new encoding